### PR TITLE
turn on musl test in CI

### PR DESCRIPTION
**Pull request by yihuaf**
Turn the musl target testing in CI. At the moment, this is only for `libcontainer` since there is an actual usecase from `aurae` project to compile libcontainer statically with `musl`. I choose to put the musl in a separate job because of the unique requirement for the nightly toolchain. While it is easy for us to switch toolchains in the dev box, it proves to be not possible to mix toolchains with the existing rust github action setup. It is easier to create a new job with the specific toolchains required.

### CODE REVIEW

File: justfile [hunk 1]
```diff
@@ -32,17 +32,21 @@ rust-oci-tests-bin:
...
+test-all: unittest test-features oci-tests containerd-test # currently not doing rust-oci here
 
 # run cargo unittests
 unittest:
     cd ./crates
     LD_LIBRARY_PATH=${HOME}/.wasmedge/lib cargo test --all --all-targets --all-features
 
 # run permutated feature compilation tests
-featuretest:
+test-features:
```
Comment (utam0k) [inline justfile:43]: 👍

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -68,7 +68,34 @@ jobs:
...
+    runs-on: ubuntu-22.04
+    timeout-minutes: 15
+    steps:
+      - uses: actions/checkout@v3
+      # We do not use `actions-rust-lang/setup-rust-toolchain` here because we
+      # want to override the default toolchain file to use nightly toolchains.
+      # The `setup-rust-toolchain` action will always choose toolchain file with
+      # no way to override. 
+      - name: Setup Rust toolchain and cache
+        uses: dtolnay/rust-toolchain@stable
```
Comment (utam0k) [inline .github/workflows/main.yml:88]: Do we need to specify `stable`? I wonder because we have had `rust-toolchain.toml`.

Comment (yihuaf) [reply .github/workflows/main.yml:88]: Good catch! I mis-read the documentation and thought the `stable` referred to the github action version... I will make the change.

**Review by utam0k — APPROVED** [review]
🚀

**Pull request merged**